easybuild/easyblocks/s/siesta.py

```diff
@@ -0,0 +1,162 @@
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Siesta, implemented as an easyblock
+
+@author: Miguel Dias Costa (National university of Singapore)
+"""
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.easyblocks.generic.makecp import MakeCp
+
+
+class EB_Siesta(MakeCp):
+    """Support for building and installing Siesta."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """
+        Define list of files or directories to be copied after make
+        """
+        extra = {
+            'files_to_copy': [[], "List of files or dirs to copy", CUSTOM],
+            'with_transiesta': [True, "Build transiesta", CUSTOM],
+            'with_utils': [True, "Build all utils", CUSTOM],
+        }
+        if extra_vars is None:
+            extra_vars = {}
+        extra.update(extra_vars)
+        return ConfigureMake.extra_options(extra_vars=extra)
+
+    def __init__(self, *args, **kwargs):
+        super(EB_Siesta, self).__init__(*args, **kwargs)
+
+        cfg_cmd = '../Src/obj_setup.sh && ../Src/configure'
+
+        if self.toolchain.options.get('usempi', None):
+            cfg_cmd += ' --enable-mpi '
+
+        build_vars = "COMP_LIBS='' "
+        build_vars += 'BLAS_LIBS="$LIBBLAS" LAPACK_LIBS="$LIBLAPACK" '
+        build_vars += 'BLACS_LIBS="$LIBBLACS" SCALAPACK_LIBS="$LIBSCALAPACK"'
+
+        self.cfg['prebuildopts'] = 'cd Obj && ' + cfg_cmd + ' && '
+        self.cfg['buildopts'] = build_vars
+
+        bins = ['Obj/siesta']
+
+        if self.cfg['with_transiesta']:
+            self.cfg['buildopts'] += ' && cd .. && mkdir Obj2 && cd Obj2 && '
+            self.cfg['buildopts'] += cfg_cmd + ' && make transiesta ' + build_vars
+
+            bins.extend(['Obj2/transiesta'])
+
+        if self.cfg['with_utils']:
+            self.cfg['buildopts'] += ' && cd ../Util && sh ./build_all.sh'
+
+            bins.extend(['Util/Bands/eigfat2plot', 'Util/Bands/new.gnubands', 'Util/Bands/gnubands'])
+
+            bins.extend(['Util/CMLComp/ccViz'])
+
+            bins.extend(['Util/COOP/mprop', 'Util/COOP/dm_creator', 'Util/COOP/fat'])
```

Comment:
I feel this can be made shorter quite a bit... There's no need for the intermittent empty lines if you structure things a bit more imho.

Try to avoid copy-pasting the same dir names (but don't take that to the extreme, using `Examples` twice for `Util/Denchar` is OK imho), and use `os.path.join`.

Also, sort binary names alphabetically (makes it easier to spot missing things or problems later).

```python
bins.extend(os.path.join('Bands', b) for b in ['eigfat2plot', 'gnubands', 'new.gnubands'])
bins.append(os.path.join('CMLComp', 'ccViz'))
bins.extend(os.path.join('COOP', b) for b in ['dm_creator', 'fat', 'mprop'])
...
self.cfg['files_to_copy'] = [([os.path.join('Utils', b) for b in bins], 'bin')]
```